integration/setup/tenant_cluster.go

```diff
@@ -97,26 +89,24 @@ func crExistsCondition(ctx context.Context, config Config) release.ConditionFunc
 			return microerror.Mask(err)
 		}
 
-		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("waited for creation of CR %#q in namespace %#q", name, crNamespace))
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("waited for creation of CR %#q in namespace %#q", id, crNamespace))
 		return nil
 	}
 }
 
-func crNotExistsCondition(ctx context.Context, config Config) release.ConditionFunc {
+func CRNotExistsCondition(ctx context.Context, id string, config Config) release.ConditionFunc {
```

Comment:
So I needed to do a lo of changes to reuse functionality while being able to use a specific tenant cluster ID for the IPAM tests. The interfaces are now maybe less nice but more flexible. Not sure if there are things we want to reconsider in here or just iterate down the road. Lets maybe just talk.

Reply:
Fine with me. I have a note to check if this condition can be generic. I'll check that later.